### PR TITLE
Working ucloud support

### PR DESCRIPTION
**Pull request by ledzep2**
Support UCloud instance routines.

### CODE REVIEW

**Comment by adieu** [comment]
Maybe we could add another centos example for ucloud later.

File: example/k8s-ucloud-ubuntu/k8s-user.yaml [hunk 1]
```diff
@@ -0,0 +1,10 @@
+apiVersion: archon.kubeup.com/v1
+kind: User
+metadata:
+  name: centos
+spec:
+  name: centos
```
Comment (adieu) [inline example/k8s-ucloud-ubuntu/k8s-user.yaml]: Why use centos as the username for an ubuntu machine?

File: example/k8s-ucloud-ubuntu/README.md [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+```
+
+Step 4
+------
+
+Modify `k8s-user.yaml`. Replace `YOUR_SSH_KEY` with your public key which will be
+used for authentication with the server. And create the user resource.
+
+```
+kubectl create -f k8s-user.yaml --namespace=k8s-centos
```
Comment (adieu) [inline example/k8s-ucloud-ubuntu/README.md]: should change k8s-centos to ucloud-ubuntu?

File: example/k8s-ucloud-ubuntu/README.md [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+Step 9
+------
+
+When the master is up, update `k8s-node.yaml` with the master ip and token and create
+it.
+
+```
+kubectl create -f k8s-node.yaml --namespace=ucloud-ubuntu
+```
+
```
Comment (adieu) [inline example/k8s-ucloud-ubuntu/README.md]: We need another step to setup flannel or calico.

File: example/k8s-ucloud-ubuntu/README.md (hunk 2 above)
Comment (adieu) [inline example/k8s-ucloud-ubuntu/README.md]: it appears to me that we didn't use k8s-user in this example.

File: example/k8s-ucloud-ubuntu/README.md (hunk 3 above)
Comment (adieu) [inline example/k8s-ucloud-ubuntu/README.md]: Also need to setup docker mirror. We could use the public one from tencent.

**Comment by adieu** [comment]
I run through the example once. Have some issued with the config yaml files. The controller worked properly though.

File: example/k8s-ucloud-ubuntu/README.md (hunk 3 above)
Comment (ledzep2) [inline example/k8s-ucloud-ubuntu/README.md]: Updated with builtin flannel installation and docker mirror.

File: example/k8s-ucloud-ubuntu/k8s-master.yaml [hunk 4]
```diff
@@ -100,7 +100,7 @@ spec:
       configs:
       - name: k8s
         data:
-          docker-mirror: https://mirror.ccs.tencentyun.com
+          docker-mirror: https://docker.mirrors.ustc.edu.cn
```
Comment (adieu) [inline example/k8s-ucloud-ubuntu/k8s-master.yaml]: something wrong with the tencent mirror?

Comment (ledzep2) [inline example/k8s-ucloud-ubuntu/k8s-master.yaml]: In my test it's not even resolvable. I assume it only works in qcloud intranet.

Comment (adieu) [inline example/k8s-ucloud-ubuntu/k8s-master.yaml]: maybe tencent changed that to intranet only. let's use the ustc one.

**Comment by adieu** [comment]
let's squash the commits and merge this one.

**Pull request merged**